invenio_rdm_migrator/streams/models/users.py

```diff
@@ -23,15 +23,28 @@ class User(Model):
     id: Mapped[int] = mapped_column(primary_key=True)
     created: Mapped[str]  # datetime
     updated: Mapped[str]  # datetime
-    username: Mapped[str]
-    displayname: Mapped[str]
+    username: Mapped[str] = mapped_column(nullable=True)
+    displayname: Mapped[str] = mapped_column(nullable=True)
     email: Mapped[str]
     password: Mapped[str]
     active: Mapped[bool]
-    updated: Mapped[str]  # datetime
+    confirmed_at: Mapped[str] = mapped_column(nullable=True)  # datetime
     version_id: Mapped[int]
+    profile: Mapped[dict] = mapped_column(JSONB(), nullable=True)
+    preferences: Mapped[dict] = mapped_column(JSONB(), nullable=True)
+
+
+class UserProfile(Model):
```

Comment:
minor: if this is used for the partial updates, I have a feeling we need to find something more versatile. We'll very likely end up in a situation where we want to partially update different subsets of a model's fields and creating a new class each time seems like overkill. 

This might require some bigger modifications in the update logic in general (i.e. right now we're passing a full ORM model for each update operation):

- either we switch to the core-style updates, i.e. `update(User.local_table).where(id=...).values(profile={...}, updated=...)`
- we adopt some sort of convention/sentinel-value for columns we don't want to update or explicitly want to set to `NULL` (e.g. `User(id=123, created=NO_UPDATE, profile=...)` or `User(id=123, updated=..., profile=NULL)`)

Reply:
Agree, we 100% need something more versatile. We cannot declare two models with the same table, therefore this code will actually fail to run.